### ISSUE TITLE
no-unreachable false positive after try/catch with conditional rethrow

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1051]
## Version

`@utoo/lint@0.2.6`

## Minimal reproduction

```js
async function render(ignoreError) {
  try {
    return await Promise.reject(new Error("render failed"));
  } catch (error) {
    if (!ignoreError) {
      throw error;
    }
  }

  return "fallback";
}

void render(true);
```

Run the rule in isolation:

```bash
utoo-lint --no-config --rules=no-unreachable --format=json repro.js
```

`@utoo/lint` reports the fallback return on line 10:

```json
{
  "line": 10,
  "column": 3,
  "severity": "warning",
  "message": "Unreachable code.",
  "ruleId": "no-unreachable"
}
```

## Expected behavior

The fallback return is reachable when `ignoreError` is `true`: the `catch` block does not rethrow and control flow continues after the `try/catch`. The same file produces no diagnostics with ESLint 8.35.0 and `no-unreachable` enabled.

## Real-world case

This false positive is reproduced in the Umi repository at [`packages/preset-umi/src/features/exportStatic/exportStatic.ts`](https://github.com/umijs/umi/blob/fb5ecefadcd1c757e506a08679f4507e1cf1d2ed/packages/preset-umi/src/features/exportStatic/exportStatic.ts#L72-L82). Umi currently keeps the rule at warning severity, but the false positive would block repositories that enable `no-unreachable` as an error.

## Environment

- macOS arm64
- Node.js 22.23.1

**Comment by fireairforce** [comment]
Verified the fix with `@utoo/lint@0.2.7`:

- the minimal reproduction now produces `diagnostics: []`
- Umi still scans 881 files successfully
- the false-positive `no-unreachable` warning in `exportStatic.ts` is gone
- Umi warning count decreased from 9 to 8 with 0 errors

The Umi integration PR has been updated to 0.2.7: https://github.com/umijs/umi/pull/13415
